mmf/datasets/multi_dataset_loader.py

```diff
@@ -189,9 +194,19 @@ def _infer_dataset_probabilities(self):
     def __len__(self):
         # Since, this is iterator, we need to return total length == number of batches
         batch_size = get_batch_size()
-        # This assumes drop_last=False for all loaders. See also
-        # build_dataloader_and_sampler().
-        return (self._total_length + batch_size - 1) // batch_size
+        # Changed the length to accomadate drop_last == True
+        # drop_last is required if the batch is split into multiple cores
+        # some of the cores may not have enough examples.
+        if is_xla():
+            logging.info(
+                "drop_last is set to True to avoid uneven dimension shapes "
+                "across cores."
+            )
+            return (self._total_length) // batch_size
+        else:
```

Comment:
Are the if and else statements changing the drop_last behavior? Should we make them consistent?

Reply:
With XLA we recommend drop_last to be true mainly to avoid uneven last batch, else the user can also incorporate dummy batches to ensure that it is not needed. In this cases I opted with droping the last batch but did not want to touch the default (GPU path). So yes case it does create a difference wrt xla and non-xla branch.